recipes/aws-c-cal/all/conanfile.py

```diff
@@ -97,5 +96,25 @@ def package_info(self):
             self.cpp_info.components["aws-c-cal-lib"].frameworks.append("Security")
         elif self.settings.os in ("FreeBSD", "Linux"):
             self.cpp_info.components["aws-c-cal-lib"].system_libs.append("dl")
+
+        self.user_info.with_openssl = self._needs_openssl
         if self._needs_openssl:
-            self.cpp_info.components["_dummy_crypto"].requires.append("openssl::crypto")
+            self.cpp_info.components["aws-c-cal-lib"].requires.append("openssl::crypto")
+            if not self.options["openssl"].shared:
+                # aws-c-cal does not statically link to openssl and searches dynamically for openssl symbols .
+                # Mark these as undefined so the linker will include them.
+                # This avoids dynamical look-up for a system crypto library.
+                crypto_symbols = [
+                    "HMAC_Update", "HMAC_Final", "HMAC_Init_ex",
+                ]
+                if tools.Version(self.deps_cpp_info["openssl"].version) >= "1.1":
+                    crypto_symbols.extend([
+                        "HMAC_CTX_new", "HMAC_CTX_free", "HMAC_CTX_reset",
+                    ])
+                else:
+                    crypto_symbols.extend([
+                        "HMAC_CTX_init", "HMAC_CTX_cleanup", "HMAC_CTX_reset",
+                    ])
+                crypto_link_flags = "-Wl," + ",".join(f"-u{symbol}" for symbol in crypto_symbols)
```

Comment:
is it portable? I doubt all linkers support `-Wl`, maybe it needs a compiler check?
and what exactly does it mean "search dynamically for openssl symbols"?
if it does `dlopen` + `dladdr` somewhere in the code, I doubt these flags may help - yes, it will link with static OpenSSL, but it will continue to look for dynamic symbols, right?

Reply:
UPD: for whose interested, the relevant code is there: https://github.com/awslabs/aws-c-cal/blob/aa89aa4950074babe84762413f39bd364ecaf944/source/unix/openssl_platform_init.c#L155

Reply:
This recipe does not support, so this recipe only supports linkers that understand the gnu ld syntax.
So I think we're good.